### ISSUE TITLE
k8s: label constraints on addons

### ISSUE THREAD

**Issue by colemickens** [Azure/acs-engine#100]
Before we can support clusters with both linux and windows nodes, we need to have os label constraints on the addons.

**Comment by anhowe** [comment]
assigning as won't fix since I use "nodeSelector: beta.kubernetes.io/os" to choose between windows and linux.

Please re-open if we should provide another way to select OS.

**Comment by rgardler** [comment]
The solution proposed only works for K8s. Are there equivalents for Swarm and Mesos?

**Comment by anhowe** [comment]
DCOS does not support windows.  For Swarm-Mode, please determine if this functionality does not exist and then file a separate issue to distinguish between Windows and Linux if it doesn't.
